### PR TITLE
Don't write playlists backup to disk on exit if not changed and add logging for files write

### PR DESCRIPTION
**Pull request by tbertels**
- Don't write playlists backup to disk on exit if not changed
Based on idea of @cmdrkotori in #255.
- Add logging for files write

### CODE REVIEW

**Comment by tbertels** [comment]
Note that because of a bug with playlist shuffling, the backup gets written every time if there's a playlist backup where shuffle is true and originalposition isn't set.

File: storage.h [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+#include <QJsonDocument>
 #include <QObject>
 
+static const char fileFavorites[] = "favorites";
+static const char fileGeometryV2[] = "geometry_v2";
+static const char fileKeys[] = "keys_v2";
+static const char filePlaylists[] = "playlists";
+static const char filePlaylistsBackup[] = "playlists_backup";
+static const char fileRecent[] = "recent";
+static const char fileSettings[] = "settings";
```
Comment (cmdrkotori) [inline storage.h:13]: Data shouldn't be in header files.  I suggest making these public static members of the Storage class, and placing their definition at the top of Storage.cpp.  main.cpp can be altered to introduce `Storage::` where it refers to these variables.

**Review by cmdrkotori — COMMENTED** [review]
Looks good so far.

File: storage.h (hunk 1 above)
Comment (tbertels) [reply storage.h]: `extern` seems like a better choice.
Those in helpers.h should be changed too.

The other ones (used locally) can be replaced by C++11 `constexpr`.

**Pull request merged**